File: Beginner/Kitchen Time Table/ktt.py
for t in range(int(input())):
        counter = 0
        n = int(input())
        a = list(map(int,input().split(" ")))
        b = list(map(int,input().split(" ")))
        for j in range(n):
            if j == 0:
                time=a[j]
            else:
                time=a[j]-a[j-1]
            if time>=b[j]:
                counter += 1
        print(counter)

File: Beginner/Kitchen Time Table/ktt.cpp
#include <iostream>
using namespace std;

int main() {
    int T, n,i,j,k,counter;
    unsigned long time;
	cin>>T;
	unsigned long *A,*B;
	for(k=0;k<T;k++) {
        counter = 0;
        cin>>n;
        A=new unsigned long[n];
        B=new unsigned long[n];
        for(i=0;i<n;i++)
            cin>>A[i];
        for(i=0;i<n;i++)
            cin>>B[i];
        for(j=0;j<n;j++) {
            if(j==0)
                time=A[j];
            else
                time=A[j]-A[j-1];
            if(time>=B[j])
                counter++;
        }
        cout<<counter<<endl;
    }
	return 0;
}
